PriscillaSam/Stackoverflow-Lite
# #160569683 Power profile page with API endpoint

#### What does this PR do?
- Create API endpoint to get user's profile. **GET** `/api/v1/users/profile`.
- Power profile page with endpoint `/api/v1/auth/login`.

#### Description of Task to be completed?
- Add API endpoint to get user's profile.
- Test endpoint using chai and mocha.
- Test endpoint on postman.
- Write fetch function to consume API endpoint


#### How should this be manually tested?
* Clone the repo using `git clone https://github.com/PriscillaSam/Stackoverflow-Lite.git`.
* Ensure that PostgreSQL is set up on your local machine. Setup your database and add your configurations to a `.env` file.
* run `git checkout ft-consume-login-api-160248060`.
* run `npm install` and `npm run start:dev`.

On Postman
* on postman, signup via `localhost:3000/api/v1/auth/login`. Get token, set it in the authorization header and send a **GET** `localhost:3000/api/v1/users/profile`.

Response body should be
```gherkin
{
    "userid": 10,
    "answers": "0",
    "asked": 0,
    "mostAnswered": [],
    "recent": []
}
```
In the browser
* in the browser, navigate to the `account.html` file and try signing up. After signup is successful, it takes you to the profile page.

#### Screenshots (if appropriate):
![success](https://user-images.githubusercontent.com/29674347/45692549-f5c76d00-bb52-11e8-9f90-dbd8754ed4d6.png)
![success](https://user-images.githubusercontent.com/29674347/45692677-44750700-bb53-11e8-9ffa-3c8c306ea432.png)


#### Any background context you want to provide?
None.

#### What are the relevant pivotal tracker stories?
#160569683 #160578248
